src/partition/partition.ts

```diff
@@ -0,0 +1,73 @@
+import { chain } from '../chain';
+import { hash } from '../hash';
+import { ifElse } from '../ifElse';
+import { reduce } from '../reduce';
+import { values } from '../values';
+
+/**
+ * @module array=>array
+ */
+/**
+ * Partition an array according a sub version of its elements.
+ * Each values in the insides arrays are identical according
+ * their form reduce by the toKey function.
+ * Comparison between 2 elements is perform with an hash of the toKey(element) form.
+ * @param toKey the function to transform array element into something that will be use to partition the array.
+ * @return the function to apply on the array to partition it according to th toKey function
+ * @example
+ * ```
+ * partition<any>(x => x.a)([{ a: 1, b: 1 }, { a: 1, b: 2 }, { a: 2, b: 1 }, { a: 2, b: 2 }])
```

Comment:
Use a valid type for documentation purpose: `partition<{ a: number; b: number; }>`

Reply:
Done